diff -Naur linux-5.12.4_old/drivers/net/usb/qmi_wwan.c linux-5.12.4_new/drivers/net/usb/qmi_wwan.c
--- linux-5.12.4_old/drivers/net/usb/qmi_wwan.c	2021-05-14 04:53:07.000000000 -0400
+++ linux-5.12.4_new/drivers/net/usb/qmi_wwan.c	2021-06-24 06:09:05.992661691 -0400
@@ -1216,6 +1216,9 @@
 	{QMI_FIXED_INTF(0x05c6, 0x9083, 3)},
 	{QMI_FIXED_INTF(0x05c6, 0x9084, 4)},
 	{QMI_FIXED_INTF(0x05c6, 0x90b2, 3)},    /* ublox R410M */
+#if 1 // Support 5G pola	0x05c6, 0x90db
+	{QMI_FIXED_INTF(0x05c6, 0x90db, 2)},    /*5G pola*/
+#endif		
 	{QMI_FIXED_INTF(0x05c6, 0x920d, 0)},
 	{QMI_FIXED_INTF(0x05c6, 0x920d, 5)},
 	{QMI_QUIRK_SET_DTR(0x05c6, 0x9625, 4)},	/* YUGA CLM920-NC5 */
