diff -Naur linux-5.12.4_old/drivers/ptp/ptp_clockmatrix.c linux-5.12.4_new/drivers/ptp/ptp_clockmatrix.c
--- linux-5.12.4_old/drivers/ptp/ptp_clockmatrix.c	2021-05-14 04:53:07.000000000 -0400
+++ linux-5.12.4_new/drivers/ptp/ptp_clockmatrix.c	2021-06-24 05:39:36.000000000 -0400
@@ -158,6 +158,15 @@
 			   u16 count)
 {
 	struct i2c_client *client = idtcm->client;
+#if 1 //Chiahsin
+	int err = 0;
+	err = i2c_smbus_read_i2c_block_data(client, regaddr, count, buf);
+	if(err < 0){
+		return err; 
+	} else {
+		return 0;
+	}
+#else
 	struct i2c_msg msg[2];
 	int cnt;
 
@@ -185,6 +194,7 @@
 	}
 
 	return 0;
+#endif
 }
 
 static int idtcm_xfer_write(struct idtcm *idtcm,
@@ -193,6 +203,15 @@
 			    u16 count)
 {
 	struct i2c_client *client = idtcm->client;
+#if 1 //Chiahsin
+	int err = 0;
+	err = i2c_smbus_write_i2c_block_data(client, regaddr, count, buf);
+	if(err < 0){
+		return err;
+	} else {
+		return 0;
+	}
+#else
 	/* we add 1 byte for device register */
 	u8 msg[IDTCM_MAX_WRITE_COUNT + 1];
 	int cnt;
@@ -213,6 +232,7 @@
 	}
 
 	return 0;
+#endif
 }
 
 static int idtcm_page_offset(struct idtcm *idtcm, u8 val)
