diff -Naur linux-5.12.4_old/drivers/leds/leds-lp5562.c linux-5.12.4_new/drivers/leds/leds-lp5562.c
--- linux-5.12.4_old/drivers/leds/leds-lp5562.c	2021-05-14 04:53:07.000000000 -0400
+++ linux-5.12.4_new/drivers/leds/leds-lp5562.c	2021-06-24 06:00:08.189012418 -0400
@@ -510,7 +510,85 @@
 	.firmware_cb        = lp5562_firmware_loaded,
 	.dev_attr_group     = &lp5562_group,
 };
+#if 1 //chiahsin
 
+static struct lp55xx_led_config lp5562_led_config[] = {
+	{
+		.name 		= "R",
+		.chan_nr	= 0,
+		.led_current	= 20,
+		.max_current	= 255,
+	},
+	{
+		.name 		= "G",		
+		.chan_nr	= 1,
+		.led_current	= 20,
+		.max_current	= 255,
+	},
+	{
+		.name 		= "B",	
+		.chan_nr	= 2,
+		.led_current	= 20,
+		.max_current	= 255,
+	},
+};
+static const u8 mode_on[] = { 0x40, 0x7F, };
+static const u8 mode_slow_blink[] = { 0x40, 0x00, 0x60, 0x00, 0x40, 0x7F, 0x60, 0x00,}; 
+static const u8 mode_fast_blink[] = { 0x40, 0x00, 0x50, 0x00, 0x40, 0x7F, 0x50, 0x00,}; 
+static const u8 mode_slow_breath[] = { 0x10, 0x7F, 0x7F, 0x00, 0x10, 0xFF, 0x7F, 0x00,};
+static const u8 mode_fast_breath[] = { 0x08, 0x7F, 0x60, 0x00, 0x08, 0xFF, 0x60, 0x00,};
+struct lp55xx_predef_pattern board_led_patterns[] = { 
+	{ 	
+		.r = mode_on,
+		.g = mode_on,
+		.b = mode_on,
+		.size_r = ARRAY_SIZE(mode_on),
+		.size_g = ARRAY_SIZE(mode_on),		
+		.size_b = ARRAY_SIZE(mode_on),    
+	},		
+	{ 	 
+		.r = mode_slow_blink,
+		.g = mode_slow_blink,
+		.b = mode_slow_blink,
+		.size_r = ARRAY_SIZE(mode_slow_blink),
+		.size_g = ARRAY_SIZE(mode_slow_blink),		
+		.size_b = ARRAY_SIZE(mode_slow_blink),  	
+	}, 
+	{ 	 
+		.r = mode_fast_blink,
+		.g = mode_fast_blink,
+		.b = mode_fast_blink,
+		.size_r = ARRAY_SIZE(mode_fast_blink),
+		.size_g = ARRAY_SIZE(mode_fast_blink),		
+		.size_b = ARRAY_SIZE(mode_fast_blink), 
+	},
+	{ 	  
+		.r = mode_slow_breath,
+		.g = mode_slow_breath,
+		.b = mode_slow_breath,
+		.size_r = ARRAY_SIZE(mode_slow_breath),
+		.size_g = ARRAY_SIZE(mode_slow_breath),		
+		.size_b = ARRAY_SIZE(mode_slow_breath),  
+	},
+	{ 	 
+		.r = mode_fast_breath,
+		.g = mode_fast_breath,
+		.b = mode_fast_breath,
+		.size_r = ARRAY_SIZE(mode_fast_breath),
+		.size_g = ARRAY_SIZE(mode_fast_breath),		
+		.size_b = ARRAY_SIZE(mode_fast_breath),   
+	},
+};
+static struct lp55xx_platform_data lp5562_platform_data = {
+        .led_config     = lp5562_led_config,
+        .num_channels   = ARRAY_SIZE(lp5562_led_config),
+#if 0 //chiahsin 		
+        .enable_gpio	= -1, 
+#endif		
+		.patterns     	= board_led_patterns,
+		.num_patterns  	= ARRAY_SIZE(board_led_patterns),
+};
+#endif 
 static int lp5562_probe(struct i2c_client *client,
 			const struct i2c_device_id *id)
 {
@@ -533,8 +611,12 @@
 			if (IS_ERR(pdata))
 				return PTR_ERR(pdata);
 		} else {
+#if 1 //chiahsin 
+			pdata = &lp5562_platform_data; 
+#else
 			dev_err(&client->dev, "no platform data\n");
 			return -EINVAL;
+#endif			
 		}
 	}
 
